pkg/snapshot/store.go

```diff
@@ -267,7 +267,28 @@ func ConfigureStore(ctx context.Context, options ConfigureStoreOptions) (*types.
 		store.Internal.Endpoint = string(secret.Data["endpoint"])
 		store.Internal.ObjectStoreClusterIP = string(secret.Data["object-store-cluster-ip"])
 		store.Internal.Region = "us-east-1"
+	} else if options.Internal && options.IsMinioDisabled {
+		clientset, err := k8sutil.GetClientset()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get k8s clientset")
+		}
+
+		if !kotsutil.IsKurl(clientset) {
+			return nil, &InvalidStoreDataError{Message: "cannot use internal storage on a non-kurl cluster"}
+		}
+
+		if store.Internal == nil {
+			store.Internal = &types.StoreInternal{}
+		}
+		store.AWS = nil
+		store.Google = nil
+		store.Azure = nil
+		store.Other = nil
+		store.FileSystem = nil
 
+		store.Provider = "replicated.com/pvc"
+		store.Bucket = "velero-internal-snapshots"
+		store.Path = "/var/velero-local-volume-provider/velero-internal-snapshots/restic"
```

Comment:
`spec.config.storageSize` is a required field, so I think we either need to check the current volume size with client-go, or we could default to 50Gb since the plugin won't change the volume size if it already exists. The volume is ALWAYS created as part of the kurl spec so the impact is minor.